src/Credentials/CredentialProvider.php

```diff
@@ -584,43 +584,57 @@ private static function loadRoleProfile($profiles, $profileName, $filename, $sts
             ? $roleProfile['role_session_name']
             : 'aws-sdk-php-' . round(microtime(true) * 1000);
 
-        if (empty($profiles[$profileName]['source_profile'])) {
-            return self::reject("source_profile is not set using profile " .
-                $profileName
+        if (
+            empty($profiles[$profileName]['source_profile'])
+            == empty($profiles[$profileName]['credential_source'])
+        ) {
+            return self::reject("Either source_profile or credential_source must be set " .
+                "using profile " . $profileName . ", but not both"
             );
         }
 
-        $sourceProfileName = $roleProfile['source_profile'];
-        if (!isset($profiles[$sourceProfileName])) {
-            return self::reject("source_profile " . $sourceProfileName
-                . " using profile " . $profileName . " does not exist"
-            );
-        }
-        $sourceRegion = isset($profiles[$sourceProfileName]['region'])
-            ? $profiles[$sourceProfileName]['region']
-            : 'us-east-1';
-
-        if (empty($stsClient)) {
-            $config = [
-                'preferStaticCredentials' => true
-            ];
-            $sourceCredentials = call_user_func(
-                CredentialProvider::ini($sourceProfileName, $filename, $config)
-            )->wait();
-            $stsClient = new StsClient([
-                'credentials' => $sourceCredentials,
-                'region' => $sourceRegion,
-                'version' => '2011-06-15',
-            ]);
+        if (
```

Comment:
Is this conditional necessary given the previous conditional with the rejection?

Reply:
Good catch, I think I initially had the logic working differently so I had to do that, but changed it to an xor in the form of an == and forgot to change it